services/wallet/router/fees/suggested_priority.go

```diff
@@ -121,18 +125,62 @@ func getEIP1559SuggestedFees(feeHistory *FeeHistory) (lowPriorityFee, avgPriorit
 	}
 
 	// Adjust low priority fee if it's equal to avg
-	lowIndex := int(float64(len(priorityFees[0])) * 0.4)
+	lowIndex := int(float64(len(priorityFees[0])) * 0.5)
 	for lowIndex > 0 && lowPriorityFee == avgPriorityFee {
 		lowIndex--
 		lowPriorityFee = priorityFees[0][lowIndex]
 	}
 
 	// Adjust high priority fee if it's equal to avg
-	highIndex := int(float64(len(priorityFees[2])) * 0.4)
+	highIndex := int(float64(len(priorityFees[2])) * 0.5)
 	for highIndex < len(priorityFees[2])-1 && highPriorityFee == avgPriorityFee {
 		highIndex++
 		highPriorityFee = priorityFees[2][highIndex]
 	}
 
 	return lowPriorityFee, avgPriorityFee, highPriorityFee, suggestedBaseFee, nil
 }
+
+func calculateNetworkCongestion(feeHistory *FeeHistory) float64 {
+	if len(feeHistory.BaseFeePerGas) == 0 || len(feeHistory.Reward) == 0 || len(feeHistory.GasUsedRatio) == 0 {
+		return 0.0
+	}
+
+	var totalBaseFee uint64
+	for _, baseFee := range feeHistory.BaseFeePerGas {
+		fee, err := hexStringToBigInt(baseFee)
+		if err != nil {
+			return 0.0
+		}
+		totalBaseFee = totalBaseFee + fee.Uint64()
+	}
+	avgBaseFee := float64(totalBaseFee) / float64(len(feeHistory.BaseFeePerGas))
+
+	var totalPriorityFee uint64
+	var countPriorityFees int
+	for _, rewardSet := range feeHistory.Reward {
+		for _, reward := range rewardSet {
+			fee, err := hexStringToBigInt(reward)
+			if err != nil {
+				return 0.0
+			}
+			totalPriorityFee = totalPriorityFee + fee.Uint64()
+			countPriorityFees++
+		}
+	}
+	avgPriorityFee := float64(totalPriorityFee) / float64(countPriorityFees)
+
+	var totalGasUsedRatio float64
+	for _, gasUsedRatio := range feeHistory.GasUsedRatio {
+		totalGasUsedRatio += gasUsedRatio
+	}
+	avgGasUsedRatio := totalGasUsedRatio / float64(len(feeHistory.GasUsedRatio))
+
+	priorityWeight := 0.7
+	gasUsedWeight := 0.3
```

Comment:
Maybe const?
Also I guess writing comment about what they are would be useful for future code reader